ecl/eclcc/NEWACTIVITY.rst

```diff
@@ -0,0 +1,537 @@
+
+  The simplest implementation of QUANTILE (sort and then select the correct rows) will always produce accurate
+  results.  However, there may be some implementations that can produce an approximate answer more quickly.  Therefore
+  we could add a SKEW attribute to allow early termination.
+
+* Does the implementation need to be stable?
+
+  In other words, if there are rows with identical values for the ordering fields, but other fields not part of the
+  ordering with different values, does it matter which of those rows are returned?  Does the relative order within those
+  matching rows matter?
+  
+  The general principle in the HPCC system is that sort operations should be stable, and that where possible activities
+  return consistent, reproducible results.  However, that often has cost - either in performance or memory consumption.
+  The design discussion highlighted that if all the fields from the row are included in the sort order then
+  the relative order does not matter because the duplicate rows will be indistinguishable.  (This is also true for
+  sorts, and following the discussion an optimization was added to 5.2 to take advantage of it.)  For the QUANTILE
+  activity we will add an ECL flag,  but also aim to spot it automatically.
+  
+* Returning counts of the numbers in each quantile might be interesting.
+
+  Possibly something to add to a future version, but not yet because there is not an immediate need.
+  
+* Is the output always sorted by the partition fields?
+
+  If this naturally falls out of the implementations then it would be worth including it in the specification.  Initially
+  we will assume not, but will revisit after it has been implemented.
+
+After the discussions we finished up with the following syntax::
+
+    QUANTILE(<dataset>, <number-of-ranges>, { sort-order } [, <transform>(LEFT, COUNTER)]
+            [,FIRST][,LAST][,SKEW(<n>)][,UNSTABLE][,SCORE(<score>)][,RANGE(set)][,DEDUP][,LOCAL]
+
+    FIRST � Match the first row in the input dataset (as quantile 0)
+    LAST �  Match the last row in the input dataset (as quantile <n>)
+    SKEW �  The maximum deviation from the correct results allowed.  Defaults to 0.
+    UNSTABLE � Is the order of the original input values unimportant?
+    SCORE � What weighting should be applied for each row.  Defaults to 1.
+    RANGE � Which quantiles should actually be returned.  (Defaults to ALL).
+    DEDUP - Avoid returning an match for an input row more than once.
+
+And a few implementation details:
+
+* The activity needs to be available in GLOBAL, LOCAL and GROUPED variants.
+* The code generator should derive UNSTABLE if no non-sort fields are returned.
+* Flags to indicate if a score/range is required.
+* Flag to indicate if a transform is required.
+
+Finally, deciding on the name of the activity took almost as long as designing it!
+
+The end result of this summarised in a JIRA issue: https://track.hpccsystems.com/browse/HPCC-12267, which contains
```

Comment:
I wasn't quite sure on what was meant by this sentence but I think you meant - 
The end result of this **_was_** summarised in **_the following_** JIRA issue:
OR
The end result of this **_surmised with the following_** JIRA issue: